final_application/remove_handwriting.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import copy
+from pathlib import Path
+from typing import Union
+
+import torch
+import torch.nn as nn
+import torchvision.transforms.functional as F
+import yaml
+from PIL import Image
+from tqdm import tqdm
+
+from inpainting_analysis_segmenter import InpaintingAnalysisSegmenter
+from models.LBAMModel import LBAMModel
+
+
+def is_image(file_name: Union[str, Path]) -> bool:
+    if not isinstance(file_name, Path):
+        file_name = Path(file_name)
+    return file_name.suffix.lower() in ['.png', '.jpeg', '.jpg', '.svg']
+
+
+def inpaint_patch(segmentation: Image, ground_truth: torch.Tensor, image_mean: bool) -> torch.Tensor:
+    # A kernel size of 7 is big enough to cover most handwriting.
+    # A padding of 3 makes sure that the image size is not different.
+    max_pooling = nn.MaxPool2d(kernel_size=7, stride=1, padding=3)
+
+    # Map the values between 0 and 1 to 0 and 1
+    segmentation_tensor = F.to_tensor(segmentation).cuda()
+    mask = torch.zeros_like(segmentation_tensor[0])
+    mask.masked_fill_(segmentation_tensor[0] >= 0.5, 1.0)
+
+    # repeat the mask, so it covers all three channel of the input image
+    # enlarge the masks so that the whole handwriting is covered
+    mask = 1 - max_pooling(mask.repeat(3, 1, 1))
+    image = ground_truth * mask
+    # the LBAM-model needs a mask as a channel of the input image
+    input_image = torch.cat((image, mask[0].unsqueeze(0)), 0)
+
+    # unsqueeze image and mask to simulate a batch of one
+    input_image = input_image.unsqueeze(0)
+    mask = mask.unsqueeze(0)
+
+    if image_mean:
+        # inpaint with image mean
+        image_mean = torch.mean(ground_truth, dim=[1, 2])
+        output = image_mean.unsqueeze(1).unsqueeze(2).repeat(1, 256, 256)
+    else:
+        # inpaint with the LBAM-model
+        netG = LBAMModel(4, 3)
+        netG.load_state_dict(torch.load('weights/inpainting.pth'))

```

Comment:
Uiuiui. Doing this inside a loop seems super inefficient. Make sure that models are only initialized once